src/Neo/Persistence/MemorySnapshot.cs

```diff
@@ -65,12 +65,22 @@ public IEnumerable<(byte[] Key, byte[] Value)> Seek(byte[] keyOrPrefix, SeekDire
 
         public byte[] TryGet(byte[] key)
         {
-            immutableData.TryGetValue(key, out byte[] value);
+            if (writeBatch.TryGetValue(key, out var value))
+            {
+                return value;
+            }
+
+            immutableData.TryGetValue(key, out value);
             return value?[..];
```

Comment:
This should return `null`? If not then change `writeBatch` to do the same.

Reply:
Yes, it returns null when it's not found

Reply:
Do you see the difference? In the returns?

Reply:
Why are we creating a new pointer?